MapCanvas.py

```diff
@@ -1004,6 +1007,24 @@ def button_release(self, event):
         self.current_selection = None
         self.curr_item_location = None
 
+    def copy_paste(self):
+        kwargs = {}
+        gimmick = self.current_gimmick
+
+        kwargs["name"] = gimmick.name
+
+        for i, name in enumerate(gimmick.param_names):
+            name_var = getattr(self, 'param{0}_name'.format(str(i)))
+            param_value1 = getattr(self, 'param{0}_value_1'.format(str(i)))
+            param_value2 = getattr(self, 'param{0}_value_2'.format(str(i)))
+
+            if param_value2.get():
+                kwargs[name_var.get()] = (param_value1.get(), param_value2.get())
```

Comment:
doesn't this code require the code in #12 to work?

Reply:
No this does not. The copy-paste standalone feature works with this code. It cannot copy *some* things, but that is a largely unrelated issue that is not related to copying and pasting itself. #12 is a minor rewrite of some of the internals of how Gimmicks are handled.

Reply:
#12 does include a slightly better version of copying and pasting that I can add onto this branch. The rest of the modified code is not really copy-paste related in itself.

Reply:
oh, sorry, I misread your code. All good :)